pxtlib/browserutils.ts

```diff
@@ -1020,7 +1019,7 @@ namespace pxt.BrowserUtils {
             return deleteDbAsync();
         return _translationDbPromise
             .then(db => db.clearAsync())
-            .catch(e => deleteDbAsync().done());
+            .catch(e => deleteDbAsync().then());
```

Comment:
>.then() [](start = 39, length = 7)

extraneous ".then()"?

Reply:
I removed the trivially extraneous `thens` in one of the last commits, but I left the one's where it's not necessarily obvious -- I'm going to file a few issues for 'further cleanup we should do' on monday. The reason I left this one (and ones like it) is that it technically changes the behavior slightly -- when it's at the end of a promise chain that is stored in a variable / returned, it clears the return value. In this case, that means if deleteDbAsync returns something for some reason (or is changed to return something) the value will not be propagated to the caller.

This is probably just being overly careful, just wanted to minimize the mental load on this pr / it's already a large one that is a lot to review.